src/content/docs/en/guides/internationalization.mdx

```diff
@@ -0,0 +1,325 @@
+
+Astro's internationalization (i18n) features allow you to adapt your project for an international audience.
+
+
+## i18n Routing (Experimental)
+
+{/* 
+
+Enable the experimental routing option by adding an `i18n` object to your Astro configuration with a default location and a list of all languages to support:
+
+```js
+// astro.config.mjs
+import {defineConfig} from "astro/config";
+
+export default defineConfig({
+    experimental: {
+        i18n: {
+            defaultLocale: "en",
+            locales: ["en", "es", "pt_BR"]
+        }
+    }
+})
+```
+
+Organize your content folders by locale, including your `defaultLocale` and `src/pages/index.astro` will now automatically default to the `index.astro` file of your default language.
+
+```
+├── src
+│   ├── pages
+│   │   ├── en
+│   │   │   ├── about.astro
+│   │   │   ├── index.astro
+│   │   ├── es
+│   │   │   ├── about.astro
+│   │   │   ├── index.astro
+│   │   ├── pt_BR
+│   │   │   ├── about.astro
+│   │   │   ├── index.astro
+│   ├── index.astro
+```
+
+Compute relative URLs for your links with `getLocaleRelativeURL` from the new `astro:i18n` module:
+
+```astro
+---
+import {getLocaleRelativeUrl} from "astro:i18n";
+const aboutUrl = getLocaleRelativeUrl("pt_Br", "about");
+---
+<p>Learn more <a href={aboutURL}>About</a> this site!</p>
+```
+
+Enabling i18n routing also provides 
+
+
+## Glossary
+
+The current page will use some specific terminology related to internationalization:
+- Locale: a code that represents a language.
+- Localized folder: a folder/directory that is named after a locale.
+
+*/}
+
+
+<Since v="3.5.0" />
+
+Astro's experimental i18n routing allows you to add your multilingual content with support for configuring a default language and URL path convention, computing relative page URLs, and accepting preferred languages provided by your visitor's browser.
+
+This routing API helps you generate, use, and verify the URLs that your multi-language site produces. Check back and update regularly for the latest changes as this API continues to develop!
+
+
+### Configure i18n routing
+
+1. Enable the experimental routing option by adding an `i18n` object to your Astro configuration with a [default location (`defaultLocale`) and a list of all languages to support (`locales`)](#defaultlocale-and-locales):
+
+   ```js title="astro.config.mjs"
+   import { defineConfig } from "astro/config"
+   export default defineConfig({
+     experimental: {
+       i18n: {
+         defaultLocale: "en",
+         locales: ["en", "es", "pt_BR"]
+       }
+     }
+   })
+   ```
+  
+    You can also specify a [`routingStrategy`](#routing-strategy) to customize the URLs for your default language.
+
+
+2. Organize your content folders with localized content by language, including your `defaultLocale`. Your folder names must match the items in `locales` exactly. `src/pages/index.astro` will now automatically redirect to the `index.astro` file of your default language.
```

Comment:
The default `routingStrategy` is `'prefix-other-locales'`, which means that the `defaultLocale` mustn't be part of the URLs.

Hence, we should suggest a different file system structure:

```
- pages
  - index.astro
  - about.astro
  - en/
    - index.astro
    - about.astro
  - pt_BR/
    - index.astro
    - about.astro
```

As a result, we don't have any redirect from `/` to `/en`.

Reply:
Can you clarify, (maybe this changed): I thought that all your FILES MUST ALWAYS be in a localized folder for both options. BUT, the URL routes generated only show the prefix for `prefix-always`.

So, it is also your FOLDER STRUCTURE that must be different for each different option?

Reply:
> So, it is also your FOLDER STRUCTURE that must be different for each different option?

Yes, that's exactly right. We don't put restrictions on where the localised folders are and their names. The routing strategy **helps** get the URLs in check without doing too many redirects, it's the responsibility of the user to have the physical pages in the correct folders.

This means that with `routingStrategy: prefix-other-locales`, a URL like this `/en/blog` will return a `404`.

I am happy to review the behaviour the strategies as we go. After all, this is experimental. I am sorry for the confusion!

Reply:
Got it! I can update to reflect this!